### PR TITLE
fix: variable onUpdate being called on same value

### PR DESCRIPTION
**Pull request by jsalaber**
# Changes

- variable `onUpdate` now should only call if the value being updated is not the same as the current value

### CODE REVIEW

File: DevCycle/DVCVariable.swift [hunk 1]
```diff
@@ -45,21 +45,23 @@ public class DVCVariable<T> {
...
+            {
+                handler(value)
+            }
             self.value = value
         } else {
             Log.warn("Variable \(variable.key) does not match type of default value \(T.self))")
         }
         self.type = variable.type
         self.evalReason = variable.evalReason
         self.isDefaulted = false
```
Comment (jonathannorris) [inline DevCycle/DVCVariable.swift:59]: pretty sure we want these values set b4 we call `handler()`

File: DevCycle/ObjC/ObjCDVCClient.swift [hunk 2]
```diff
@@ -136,8 +136,8 @@ public class ObjCDVCClient: NSObject {
         return variable(key: key, defaultValue: defaultValue)
     }
     
-    @objc public func boolVariable(key: String, defaultValue: Bool) -> ObjCDVCVariable {
-        return variable(key: key, defaultValue: defaultValue)
+    @objc public func boolVariable(key: String, defaultValue: NSNumber) -> ObjCDVCVariable {
+        return variable(key: key, defaultValue: defaultValue.boolValue)
```
Comment (jonathannorris) [inline DevCycle/ObjC/ObjCDVCClient.swift:140]: why is this change needed? can we do the opposite where we cast the `Bool` to a `NSNumber` in here?

File: DevCycle/Utils/isEqual.swift [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+        return lhs == rhs
+    } else if let lhs = lhs as? Float, let rhs = rhs as? Float {
+        return lhs == rhs
+    } else if let lhs = lhs as? Int, let rhs = rhs as? Int {
+        return lhs == rhs
+    } else if let lhs = lhs as? String, let rhs = rhs as? String {
+        return lhs == rhs
+    } else if let lhs = lhs as? Bool, let rhs = rhs as? Bool {
+        return lhs == rhs
+    }
```
Comment (jonathannorris) [inline DevCycle/Utils/isEqual.swift:23]: should there be an `else` with a warning that the type is not known?

**Review by jonathannorris — CHANGES_REQUESTED** [review]

File: DevCycle/Utils/isEqual.swift [hunk 4]
```diff
@@ -0,0 +1,25 @@
+//
+//  isEqual.swift
+//  DevCycle
+//
+//  Copyright © 2022 Taplytics. All rights reserved.
+//
+
+import Foundation
+
+public func isEqual<T>(_ lhs: T, _ rhs: T) -> Bool {
```
[the remark this answers is not in the document]
Comment (jsalaber) [reply DevCycle/Utils/isEqual.swift:10]: left hand side, right hand side. It's used commonly as param names for operators on two operands

File: DevCycle/DVCVariable.swift (hunk 1 above)
Comment (jonathannorris) [reply DevCycle/DVCVariable.swift:59]: @jsalaber this one still needs to be fixed.

Comment (jsalaber) [reply DevCycle/DVCVariable.swift]: fixed
